api_blogs/lib/api_blogs/blog.ex

```diff
@@ -103,6 +120,11 @@ defmodule ApiBlogs.Blog do
     User.changeset(user, attrs)
   end
 
+  def extract_id(conn) do
+    conn.private[:guardian_default_token]
+    |> Guardian.decode_and_verify()
+  end
```

Comment:
O que acha que fazer assim ? Evita o `pipe` e já extrair direto com `pattern matching`
 
```suggestion
  def extract_id(%{private: {guardian_default_token: guardian_default_token}}) do
    Guardian.decode_and_verify(guardian_default_token)
  end
```

Reply:
Um outro ponto bacana dessa sugestão do Caciquinho é que garante que o campo `guardian_default_token` esteja presente no `map`.

A abordagem de acesso dinâmico `conn.private[:guardian_default_token]` pode retornar `nil` tanto quando a chave não existe no `map` quanto quando o valor da chave é de fato `nil`.

Reply:
Vou fazer essa mudança, obrigada pela sugestão!